### PR TITLE
Reset_Password

### PR DESCRIPTION
**Pull request by ngyihkai**
Ready

### CODE REVIEW

**Comment by ngyihkai** [comment]
Ready

File: app/models/user.rb [hunk 1]
```diff
@@ -1,6 +1,5 @@
 class User < ApplicationRecord
-  has_secure_password
```
Comment (ngocvt-0484) [inline app/models/user.rb:2]: why you delete

File: app/models/user.rb [hunk 2]
```diff
@@ -23,6 +22,20 @@ def User.digest string
       BCrypt::Password.create string, cost: cost
     end
   end
+
+  def create_reset_digest
+      self.reset_token = User.new_token
```
Comment (ngocvt-0484) [inline app/models/user.rb:27]: wrong indent

File: app/models/user.rb (hunk 1 above)
Comment (ngyihkai) [reply app/models/user.rb:2]: because i have 2 has_secure_password, so i remove one of it.

**Comment by ngyihkai** [comment]
Ready

File: app/mailers/user_mailer.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  # def account_activation(user)
+  #   @user = user
+  #   mail to: user.email, subject: "Account activation"
+  # end
+  # Subject can be set in your I18n file at config/locales/en.yml
+  # with the following lookup:
+  #
+  #   en.user_mailer.password_reset.subject
+  #
+  def password_reset
```
Comment (huongnt-2545) [inline app/mailers/user_mailer.rb:17]:
```suggestion
  def password_reset user
```

**Comment by ngyihkai** [comment]
Ready

**Pull request merged**